### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by georg3003** [jordi-pujol/wwanHotspot#18]
Hello,
I found this repository by browsing openwrt, looking for the problem when a AP is not working when there is no client wifi connection for internet.
So I tried this, and it's working really well. Changing the client wifi ssid for different ssid's. 

Now I see, that there is still no AP activated when there is no wifi client active. 
And I still need to investigate, if on reboot it is still connecting or not, if wwanhotspot is coming up. It seems yes, but if there is no wifi available there is also no AP activated. So I can't connect with my openwrt router. 

My question is now, if you can optimize the activation after reboot, so even if there is no client wifi to connect to activate the AP's so I can connect to the local network. 
It seems, after first connection of wifi client and disconnecting, AP's are still accessible(local network present) 

OH and second, is there a possibility to connect to an unknown open wifi and test if internet is active and establish a VPN (ipsec, openvpn or the new VPN wire...) for secure access to the internet through the unsecured open wifi connection.

Is there also a built in priority for which wifi network to use first?
Thank you very much for this really good peace of software. 
Best regards
Georg

**Comment by jordi-pujol** [comment]
> Now I see, that there is still no AP activated when there is no wifi client active.
This is not the behavior of wwanHotspot, AP must be allways activated, your configuration must have some problems. We must debug this case to find the problem, please check your configuration.

> OH and second, is there a possibility to connect to an unknown open wifi and test if internet is active and establish a VPN (ipsec, openvpn or the new VPN wire...) for secure access to the internet through the unsecured open wifi connection.
According to the wifi standards, to connect to any wifi network an SSID must be allways specified This makes me think in a new feature of wwanHotspot that allows to specify SSID wilcards and wwanHotspot will check for it when scanning, 

> Is there also a built in priority for which wifi network to use first?
No, now wwanHotspot connects to the available Hotspots by rotation.

Regards,
Jordi Pujol

**Comment by georg3003** [comment]
Hi,
thank you for your help.


> > Now I see, that there is still no AP activated when there is no wifi client active.
> This is not the behavior of wwanHotspot, AP must be allways activated, your configuration must have some problems. We must debug this case to find the problem, please check your configuration.

I first stage I changed nothing except the Additional wifi. It worked partially . So far so good. I corrected the wifi config to netX_..... and I remarked it's not working, so I put it back to normal. Now its working again.
BUT
After initial start with existing wifi its working. But if at startup is no wifi, the daemon will not start. I tried to make a cron command every minute then every 3 minutes to start or restart the daemon. but no success 
`* or "*/3 * * * * /etc/init.d/wwanHotspot restart" or start`
is not restarting the daemon. on ssh /etc/init.d/wwanHotspot restart will restart and it seems that its working.

**Howcan I do a modification to the script, to activate the daemon, without deactivating itself after first scan.**

Now I did to my openwrt the following modification : I disabled the Wifi client only in Lucy, but the wwanhotspot daemon starts and connects to the wifi"server" It's a little bit weird but it works fine. It's working, but Lucy shows, wifi client is deactivated with the correctly modified  hotspot name. That's really good. ;-)


> 
> > OH and second, is there a possibility to connect to an unknown open wifi and test if internet is active and establish a VPN (ipsec, openvpn or the new VPN wire...) for secure access to the internet through the unsecured open wifi connection.
> According to the wifi standards, to connect to any wifi network an SSID must be allways specified This makes me think in a new feature of wwanHotspot that allows to specify SSID wilcards and wwanHotspot will check for it when scanning,

Sorry, but I know that wpasupplicant can be configured to connect to any open wifi network (SSID). Thats why I suggested this.

> > if internet is active and establish a VPN
> like an option netX_OnConnect execute a command ?

Sorry I am not so familiar with uci commands on openwrt. yet. :-(    I need to read much more. But I think yes.

....

> Regards,
> Jordi Pujol



So let try to me explain once more how I am doing my "start". 

First scenario:
1. My server wifi is on
2. start my openwrt avm4040
3. waiting, a little bit, and after reboot its connected.

Second scenario:
1. My server wifi is off
2. start my openwrt avm4040
3. waiting, 5 minutes for example.
4. switching on server wifi
5. waiting again, no connection.
6. on ssh "/etc/init.d/wwanHotspot restart" will connect 

Here your text

`Automatic scanning for available hotspots.

**At start wwanHotspot will look once for a Hotspot**. Also will look for availability of another Hotspot after disconnection.

We may configure conditionally scanning:

scan automatically only when there is no default route to external networks
allways scan for hotspots at every time interval
scan only when requested externally`

An "ALLWAYS" also does not help, in this situation.

so in conclusion:
**Howcan I do a modification to the script, to activate the daemon, without deactivating itself after first scan. 
But better is to stay scanning every 30/60/90/120 Seconds if there is no said connected**

Thank you for your advice 
Best regards
Georg

**Comment by jordi-pujol** [comment]
all Ok, thanks

**Comment by georg3003** [comment]
I need help,

but you did not respond my Question.

Please:
How can I do a modification to the script, to activate the daemon, without deactivating itself after first scan?

**Comment by jordi-pujol** [comment]
please read the github rules and try to understand why you should not do a question like this,
and also read carefully your previous message and ask someone if he has understood a single bit ot it.
